src/System.Private.CoreLib/shared/System/Collections/ObjectModel/Collection.cs

```diff
@@ -125,12 +125,12 @@ public void InsertRange(int index, IEnumerable<T> collection)
 
             if (collection == null)
             {
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.list);
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.collection);
             }
 
             if ((uint)index > (uint)items.Count)
```

Comment:
Nit: It may be nice to switch all places in this file that do `if (index < 0 || index >= items.Count)` or `if (index < 0 || index > items.Count)` to this form of error handling while you are on it.

Reply:
Something I was thinking about submitting another PR for, but we can do it in this one. I'll update it right now

Reply:
There is one more in `public T this[int index]`

Reply:
This is a good catch, fixing it now